crates/oxc_transformer/tests/integrations/plugins/replace_global_defines.rs

```diff
@@ -68,7 +67,22 @@ fn dot_with_overlap() {
     ])
     .unwrap();
     test("import.meta.env", "__foo__", config.clone());
-    test("import.meta.env.NODE_ENV", "import.meta.env.NODE_ENV", config.clone());
+    test("import.meta.env.FOO", "import.meta.env.FOO", config.clone());
+
+    test(
+        r"
+console.log(
+	[import.meta.env.FOO = 0]
+)
+        ",
+        "
+
+console.log(
+	[import.meta.env.FOO = 0]
+)
+",
+        config.clone(),
+    );
```

Comment:
This test seems overly complicated, and doesn't cover all cases.

```suggestion
    test("import.meta.env = 0", "__foo__ = 0");
    test("import.meta.env.NODE_ENV = 0", "__foo__.NODE_ENV = 0");
    test("import.meta.env.FOO = 0", "import.meta.env.FOO = 0");
```

Reply:
Yes